Automattic/newspack-ios
# Issues/91 sorting settings

Closes #91 

In retrospect, I wish I'd have separate this PR into two, one related to sorting and one related to the last of the syncing changes... but here we are. 😛 

This is the final (planned) PR for the Photos feature.  It does the following:
- Adds a Story Folder setting to automatically sync (or not) assets as they are added. 
- Updates assets so they can be flagged for upload if needed.
- Appropriate UI changes for both.
- The cloud icons in cells now only appear when an asset would not automatically be synced.
- Removed cloud icons from the Story and Assets controller nav bars.
- Removes the manual ordering functionality of assets. This was wonky and we're not sure it has much real world use.
- Modifies sorting of the Asset and Folders lists for consistency.

To Test:
Start from a clean build.  The changes to sorting and database could create conflicts otherwise.

Scenario 1: Smoke test for regression.
- Create a few Story Folders and add some assets to them.  Confirm syncing / sharing works as expected and there is no regression.  By default story folders should automatically sync assets (the previous behavior).

Scenario 2: Manual Upload.
- Edit a folder, or create a new folder and set toggle it's sync behavior. 
- Add some new photo assets. Confirm the photos do not automatically upload.
- Confirm the photo assets show the cloud icon.
- Pull to refresh on the story of the photos.  Confirm photos do not automatically upload.
- Tap one of the cloud icons.  Confirm the photo is uploaded and the others are not.

Scenario 3: Sort order
- Smoke test sorting folders by name ascending / descending. 
- Smoke test sorting folders by date. (Note if the folders are created at the same time the sorting by date might not change the order.)
- Smoke test sorting assets by type ascending / descending.
- Date sorting for assets is by day.  Confirm that assets show up in a single section for the day. 

@jleandroperez could I trouble you for this last one for Photos?
